src/skills-builder/skills-builder-modal/view-results/ViewResults.jsx

```diff
@@ -1,7 +1,81 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
+import {
+  Stack, Row, Alert, Spinner,
+} from '@edx/paragon';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { CheckCircle } from '@edx/paragon/icons';
+import { SkillsBuilderContext } from '../../skills-builder-context';
+import RelatedSkillsSelectableBoxSet from './RelatedSkillsSelectableBoxSet';
+import { searchJobs, getProductRecommendations } from '../../utils/search';
+import messages from './messages';
 
-const ViewResults = () => (
-  <h3>Results will render on this step</h3>
-);
+const ViewResults = () => {
+  const { algolia, state } = useContext(SkillsBuilderContext);
+  const { jobSearchIndex, productSearchIndex } = algolia;
+  const { careerInterests } = state;
+
+  const [selectedJobTitle, setSelectedJobTitle] = useState('');
+  const [jobSkillsList, setJobSkillsList] = useState([]);
+  // eslint-disable-next-line no-unused-vars
+  const [courseRecommendations, setCourseRecommendations] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const getJobs = async () => {
+      // fetch list of jobs with related skills
+      const jobInfo = await searchJobs(jobSearchIndex, careerInterests);
+
+      // fetch course recommendations based on related skills for each job
+      const results = await Promise.all(jobInfo.map(async (job) => {
+        const formattedSkills = job.skills.map(skill => skill.name);
+
+        const recommendations = await getProductRecommendations(productSearchIndex, 'course', formattedSkills);
```

Comment:
IIRC, in the B2B version, after retrieving the course recommendations, they have some logic to de-dupe courses the learner is already enrolled in. Will we be covering this somewhere for course recommendations?

Reply:
Another good catch! This will be covered in the next ticket. It looks like they are [using an API in the LMS](https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/master/src/components/skills-quiz/data/service.js#L28-L32) to retrieve the information on enrollments. As far as I can tell, we should be able to do something pretty similar?

Reply:
Yep, absolutely! Thanks, Max.